native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -124,6 +127,71 @@ public static BStream nativeQuery(
         }
     }
 
+    public static Object nativeQueryRow(
+            BObject client, Object paramSQLString,
+            BTypedesc ballerinaType,
+            AbstractStatementParameterProcessor statementParameterProcessor,
+            AbstractResultParameterProcessor resultParameterProcessor) {
+        Object dbClient = client.getNativeData(Constants.DATABASE_CLIENT);
+        TransactionResourceManager trxResourceManager = TransactionResourceManager.getInstance();
+        if (dbClient != null) {
+            SQLDatasource sqlDatasource = (SQLDatasource) dbClient;
+            if (!((Boolean) client.getNativeData(Constants.DATABASE_CLIENT_ACTIVE_STATUS))) {
+               return ErrorGenerator.getSQLApplicationError("SQL Client is already closed, hence " +
+                        "further operations are not allowed");
+            }
+            Connection connection = null;
+            PreparedStatement statement = null;
+            ResultSet resultSet = null;
+            String sqlQuery = null;
+            try {
+                if (paramSQLString instanceof BString) {
+                    sqlQuery = ((BString) paramSQLString).getValue();
+                } else {
+                    sqlQuery = Utils.getSqlQuery((BObject) paramSQLString);
+                }
+                connection = SQLDatasource.getConnection(trxResourceManager, client, sqlDatasource);
+                statement = connection.prepareStatement(sqlQuery);
+                if (paramSQLString instanceof BObject) {
+                    statementParameterProcessor.setParams(connection, statement, (BObject) paramSQLString);
+                }
+                resultSet = statement.executeQuery();
+                if (!resultSet.next()) {
+                    return ErrorGenerator.getNoRowsError("Query did not retrieve any rows.");
+                }
+
+                Type describingType = ballerinaType.getDescribingType();
+
+                // If the return data type is a record
+                if (describingType.getTag() == TypeTags.RECORD_TYPE_TAG) {
```

Comment:
What if the check keyword is not used, here the returning type is a union of record and error

Reply:
Works as intended. If the query returns a record, the method returns successfully. Else throws an error.

Added test to verify in 91e3f85